libcudacxx/include/cuda/std/__cmath/isnan.h

```diff
@@ -152,14 +153,18 @@ template <class _Tp>
 #endif // _CCCL_HAS_NVFP4_E2M1()
 
 #if _CCCL_HAS_FLOAT128()
+
 [[nodiscard]] _CCCL_API constexpr bool isnan(__float128 __x) noexcept
 {
-#  if defined(_CCCL_BUILTIN_ISNAN)
-  return _CCCL_BUILTIN_ISNAN(__x);
-#  else // ^^^ _CCCL_BUILTIN_ISNAN ^^^ / vvv !_CCCL_BUILTIN_ISNAN vvv
-  return _CUDA_VSTD::__isnan_impl(__x);
-#  endif // ^^^ !_CCCL_BUILTIN_ISNAN ^^^
+#  if _CCCL_HAS_FLOAT128_CUDA_FUNCTIONS()
+  if (!_CUDA_VSTD::__cccl_default_is_constant_evaluated())
+  {
+    NV_IF_TARGET(NV_PROVIDES_SM_100, (return ::__nv_fp128_isnan(__x);));
+  }
+#  endif
+  return __x != __x;
```

Comment:
I do not see any difference in [generated SASS](https://godbolt.org/z/GjPY4f6TE) for this

What do we need the special case for?